### PR TITLE
Observers can auto-orbit meteors; space dust event

### LINKED ISSUE #19137: Meteors never move
Turns out that addtimer doesn't agree with `walk_towards`

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: During a meteor shower, observers can automatically orbit
threatening meteors via an action button and watch them hit the station.
rscadd: Added Major Space Dust event, which is a meteor shower
containing only space dust.
fix: Fixes bug where meteors wouldn't move when spawned.
add: Meteor Shower split into three different events, Normal, Threatening and Catastrophic. They all have the same crew messages, but admins can tell the difference.
/:cl:

Fixes #19137.
- Adds the new SSaugury subsystem for holding the state of all the interested ghosts
  wanting to watch heavy things hit the station. @MrStonedOne for advice on suitable priority
  numbers and flags for this new subsystem.
- Also, as always, a bunch of random unrelated minor changes in files that I came across.
- Giant tunguska meteors now trail purple particles to distinguish them visually from flaming meteors.
- Meteor total weights have gone up, but Major Space Dust overal lowers their chance of being picked. Also increase Meat wave.
- Meteors auto delete after 3 minutes, to stop meteors being stuck in a loop indefinitely.
- Potential future extensions: Making threat determined by whether the meteor will actually hit the station or not. Making meteor movement not use a really super old byond proc that no one likes.

### CODE REVIEW

File: code/controllers/subsystem/augury.dm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	var/list/observers_given_action = list()
+
+/datum/subsystem/augury/New()
+	NEW_SS_GLOBAL(SSaugury)
+
+/datum/subsystem/augury/stat_entry(msg)
+	..("W:[watchers.len]|D:[doombringers.len]")
+
+/datum/subsystem/augury/proc/register_doom(atom/A, severity)
+	doombringers[A] = severity
```
Comment (ChangelingRain) [inline code/controllers/subsystem/augury.dm]: did you do all of this for these names

Comment (coiax) [inline code/controllers/subsystem/augury.dm]: I didn't really know what to call them, honest.

File: code/controllers/subsystem/augury.dm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	for(var/w in watchers)
+		if(!w)
+			watchers -= w
+			continue
+		var/mob/dead/observer/O = w
+		if(O.orbiting)
+			continue
+		else if(biggest_doom)
+			O.orbit(biggest_doom)
```
Comment (MrStonedOne) [inline code/controllers/subsystem/augury.dm]: reminder that orbit blocks.

File: code/controllers/subsystem/augury.dm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			doombringers -= d
+			continue
+		var/threat = doombringers[d]
+		if((biggest_threat == null) || (biggest_threat < threat))
+			biggest_doom = d
+			biggest_threat = threat
+
+	if(doombringers.len)
+		for(var/i in player_list)
+			if(isobserver(i) && (!(i in observers_given_action)))
```
Comment (RemieRichards) [inline code/controllers/subsystem/augury.dm]: ```if(isobserver(i) && (!(i in observers_given_action)))``` should be ```if(isobserver(i) && observers_given_action[i])```

File: code/controllers/subsystem/augury.dm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+				var/datum/action/innate/augury/A = new
+				A.Grant(i)
+				observers_given_action += i
```
Comment (RemieRichards) [inline code/controllers/subsystem/augury.dm]: ```observers_given_action += i``` -> ```observers_given_action[i] = 1```

File: code/game/gamemodes/meteor/meteors.dm [hunk 5]
```diff
@@ -160,16 +165,18 @@
...
 	for(var/throws = dropamt, throws > 0, throws--)
 		var/thing_to_spawn = pick(meteordrop)
 		new thing_to_spawn(get_turf(src))
 
-/obj/effect/meteor/proc/meteor_effect(sound=1)
-	if(sound)
+/obj/effect/meteor/proc/meteor_effect()
+	if(heavy)
 		for(var/mob/M in player_list)
+			if((M.orbiting) && (M in SSaugury.watchers))
```
Comment (RemieRichards) [inline code/game/gamemodes/meteor/meteors.dm]: ```SSaugury.watchers[M]```

File: code/controllers/subsystem/augury.dm [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		else if(biggest_doom)
+			addtimer(O, "orbit", 0, FALSE, biggest_doom)
+
+/datum/action/innate/augury
+	name = "Auto Follow Debris"
+	icon_icon = 'icons/obj/meteor.dmi'
+	button_icon_state = "flaming"
+
+/datum/action/innate/augury/Activate()
+	SSaugury.watchers |= owner
```
Comment (RemieRichards) [inline code/controllers/subsystem/augury.dm]: ```SSaugury.watches[owner] = 1```

**Pull request merged**